dbms/src/Storages/Page/V3/BlobStore.cpp

```diff
@@ -418,11 +430,22 @@ PageMap BlobStore::read(FieldReadInfos & to_read, const ReadLimiterPtr & read_li
         [](const FieldReadInfo & a, const FieldReadInfo & b) { return a.entry.offset < b.entry.offset; });
 
     // allocate data_buf that can hold all pages with specify fields
+
     size_t buf_size = 0;
     for (auto & [page_id, entry, fields] : to_read)
     {
         (void)page_id;
-        buf_size += entry.size;
+        // Sort fields to get better read on disk
+        std::sort(fields.begin(), fields.end());
+        for (const auto field_index : fields)
+        {
+            buf_size += entry.getFieldSize(field_index);
+        }
```

Comment:
Add related `ASSERT_*` on the ut `BlobStoreTest`

Reply:
https://github.com/pingcap/tiflash/pull/4691/files#diff-d56b8440bd7f12b645bfea098e6c00fee6a6fc169932d5eb1a9150a2f9a95adb already added.